lte/gateway/c/core/oai/tasks/amf/amf_smf_send.cpp

```diff
@@ -593,7 +595,7 @@ M5GSmCause amf_smf_get_smcause(amf_ue_ngap_id_t ue_id, ULNASTransportMsg* msg) {
   the external DNN because the DNN was not included
   although required or if the DNN could not be resolved.
   */
-  if (msg->dnn.len == 0 &&
+  if (msg->dnn.len <= 1 &&
```

Comment:
what is the reason for this change? should not this be instead `msg->dnn.len == 0 || (ue_context->amf_context.apn_config_profile.nb_apns == 0)`

Reply:
![MicrosoftTeams-image (5)](https://user-images.githubusercontent.com/89978170/143176796-9c777fc6-23c0-43ae-b50a-be708740aea9.png)
msg->dnn.len is the length of dnn contents. DNN value will have the first byte that tells the DNN length and then DNN value. In this case msg->dnn.len is 1 means no DNN content. Minimum value should be more than 1. 

```should not this be instead msg->dnn.len == 0 || (ue_context->amf_context.apn_config_profile.nb_apns == 0)```
If UE sends empty DNN or DNN IE not present. As per PR #9645 default DNN is picking from apn_config_profile.

Reply:
what about the AND vs OR statement here. Should not it be OR? If there is no apn profile for UE, it should be rejected.

Reply:
If there is no apn profile for UE and UE sends DNN. As per PR #9645 PDU session will be rejected with cause 91: DNN not supported or not subscribed.

Reply:
![image](https://user-images.githubusercontent.com/89978170/143464589-08b11c50-2c17-47a5-83ad-6a863834fd07.png)
@ulaskozat these are the cases we are handling and their current behavior. Please review it and provide the feedback. 
Above the case is for the Cause 27, #9645 PR handling other scenarios.

Reply:
Cause #91 was implemented in ```lte/gateway/c/core/oai/tasks/amf/amf_smf_send.cpp``` in PR #9645.
![image](https://user-images.githubusercontent.com/89978170/143985732-adbb8304-244c-45a1-bf4c-2f82cde8493c.png)
![image](https://user-images.githubusercontent.com/89978170/143985791-bde35b19-a492-4c3d-af4e-65912cf32ce3.png)